data-products/src/article_text_streaming/article_text_streaming_flow.py

```diff
@@ -250,9 +250,19 @@ def handle_html2text(html: str, key: str, resolved_id: int):
 
     fileobj = BytesIO(contents)
     with gzip.GzipFile(fileobj=fileobj) as gzipfile:
-        contents_str = gzipfile.read()
-    dicts = [json.loads(c) for c in contents_str.splitlines()]
-    df = pd.DataFrame.from_records(dicts)
+        contents_str = gzipfile.read().decode("utf-8")
+
+    records = []
+    for line in contents_str.splitlines():
+        if not line.strip():
+            continue  # Ignore empty lines. On 2025/02/26 a file mysteriously had an empty line between records.
+        try:
+            record = json.loads(line)
+            records.append(record)
+        except json.JSONDecodeError as e:
+            logger.warning(f"Invalid JSON in blob {key}: {line[:100]}... Error: {e}")
```

Comment:
I have not observed invalid JSON, but I thought I might as well guard against that, too. I can remove it if you prefer a fail-fast approach.